pkg/util/helper/work_test.go

```diff
@@ -94,16 +353,108 @@ func TestGenEventRef(t *testing.T) {
 			wantErr: true,
 		},
 	}
+
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			actual, err := GenEventRef(tt.obj)
-			if (err != nil) != tt.wantErr {
-				t.Errorf("GenEventRef() error = %v, wantErr %v", err, tt.wantErr)
-				return
-			}
-			if !reflect.DeepEqual(actual, tt.want) {
-				t.Errorf("GenEventRef() = %v, want %v", actual, tt.want)
+			if tt.wantErr {
```

Comment:
https://github.com/karmada-io/karmada/blob/5140eb1e1c757c782d54a60a57b19947e0391848/pkg/util/helper/work.go#L131-L160

The function `TestGenEvenRef` does not cover scenarios where `resource.Kind is empty or `resource.Name` is empty. Could you add these cases? This would improve the test coverage.

Reply:
Sure, Thanks!